src/helpers/spacewalk.ts

```diff
@@ -35,12 +35,11 @@ export function convertCurrencyToStellarAsset(currency: SpacewalkPrimitivesCurre
     if (stellarAsset.isStellarNative) {
       return Asset.native();
     } else if (stellarAsset.isAlphaNum4) {
-      const code = hex_to_ascii(stellarAsset.asAlphaNum4.code.toHex());
-      console.log('code', code);
+      const code = hex_to_ascii(stellarAsset.asAlphaNum4.code.toHex()).trim();
       const issuer = convertRawHexKeyToPublicKey(stellarAsset.asAlphaNum4.issuer.toHex());
       return new Asset(code, issuer.publicKey());
     } else if (stellarAsset.isAlphaNum12) {
-      const code = hex_to_ascii(stellarAsset.asAlphaNum12.code.toHex());
+      const code = hex_to_ascii(stellarAsset.asAlphaNum12.code.toHex()).trim();
```

Comment:
`trim()`ing is problematic because we then support exactly those assets that are 'faulty'. 

We will soon have the assets `BRL `, `TZS ` and `BRL\0`, `TZS\0` on-chain. 
What we want to do is, ignore `BRL `, `TZS ` (ie. the assets with a space character) and support `BRL\0`, `TZS\0`. Thus, instead of using `trim()` we should instead `replace('\0','')` if necessary. We should also reject all asset codes that contain space characters, just to be on the safe side. In these cases we can just return `null`.

Reply:
OHH. Okay it was a misunderstanding. Thanks for clarifying that :)